src/library.js

```diff
@@ -2386,6 +2386,32 @@ addToLibrary({
   _emscripten_get_now_is_monotonic__deps: ['$nowIsMonotonic'],
   _emscripten_get_now_is_monotonic: () => nowIsMonotonic,
 
+#if !SHARED_MEMORY
+  __emscripten_atomics_sleep__internal: true,
+  // In a browser without cross origin isolation, SharedArrayBuffer is deleted
+  // from the global scope:
+  // https://html.spec.whatwg.org/multipage/webappapis.html#creating-a-new-javascript-realm
+  // However, it says:
+  //
+  // > Web developers can still get at the constructor through
+  // > `new WebAssembly.Memory({ shared:true, initial:0, maximum:0}).buffer.constructor`.
+  //
+  // That is what we are up to here. Use an IIFE to avoid shadowing
+  // SharedArrayBuffer globally.
+  __emscripten_atomics_sleep__postset: `
+    var SABConstructor = new WebAssembly.Memory({"shared":true,"initial":0,"maximum":0}).buffer.constructor;
+    var waitBuffer = new Int32Array(new SABConstructor(4));
+    try {
+      __emscripten_atomics_sleep(0);
+      Module._supports_atomics_wait = true;
```

Comment:
Instead of storing publicly on the `Module` object, could this be changed to an internal variable? Adding `$supportsAtomicWait: false` to the library.js above should do it.

Reply:
Alternatively, you could change `_emscripten_thread_supports_atomics_wait` to be `() => false` by default and reassign to `() => true` here to avoid a separate variable.

Reply:
I think you could even just declare `var supports_atomics_wait` here since those code is running in the top level/module scope.  (ie.. just removing the `Module` prefix should be enough.)

Reply:
Ah right, because it's postset. Yeah that would work too.

Reply:
Okay I did that.